### ISSUE TITLE
Two repositories were bound to the same sidecar on the same port via cyral_repository_binding

### ISSUE THREAD

**Issue by greenkiwi** [cyralinc/terraform-provider-cyral#146]
I applied roughly the configuration here:
```
resource "cyral_sidecar" "sidecar_1" {
  name              = "sidecar-1"
  deployment_method = "helm3"
}

resource "cyral_sidecar" "sidecar_2" {
  name              = "sidecar-2"
  deployment_method = "helm3"
}

resource "cyral_repository" "repo_1" {
  name = "repo_1"
  host = var.database.address
  port = var.database.port
  type = "mysql"
}

resource "cyral_repository_binding" "repo_1" {
  repository_id = cyral_repository.repo_1.id
  listener_host = "0.0.0.0"
  listener_port = 3307
  sidecar_id    = var.sidecar_1_id
}

resource "cyral_repository" "repo_2" {
  name = "repo_2"
  host = var.database.address
  port = var.database.port
  type = "mysql"
}

resource "cyral_repository_binding" "repo_2" {
  repository_id = cyral_repository.repo_2.id
  listener_host = "0.0.0.0"
  listener_port = 3307
  sidecar_id    = var.sidecar_1_id
}
```

Note that both `repo_1` and `repo_2` bindings are set to bind pory 3307 and the `repo_2` had a typo where it was being attached to `sidecar-1`.

## Expected:
I would have expected this to error out when it attempted to add a second binding with the same port.

## Actual

Sidecar 1 had both repositories bound to port `3307`

![image](https://user-images.githubusercontent.com/54169/145741432-1e9eec0d-b5c4-4683-a086-7ca35bbb1553.png)

This created issues because I couldn't then disable the repos to remove them via the UI and had to delete the sidecar.

**Comment by wcmjunior** [comment]
The issue is caused by a bug in the control plane backend. The fix is coming in control plane `v2.31`. Cyral team: refer to internal tickets `ENG-7521` and `ENG-8350`.

**Comment by wcmjunior** [comment]
@greenkiwi , the fix will be available in `v2.31`, but the release is still in the oven. Please talk to customer success team to get a sense on when you can get this upgrade. Thanks again for all the inputs and amazing contributions towards the improvements of our product. :-)
